config.js

```diff
@@ -52,14 +52,14 @@ config = Object.assign({}, config, {
   // ----------------------------------
   // Compiler Configuration
   // ----------------------------------
-  compiler_devtool: false,
-  compiler_hash_type: 'hash',
+  compiler_devtool: __DEV__ && 'eval-cheap-module-source-map' || __STAGING__ && 'source-map',
+  compiler_hash_type: __PROD__ ? 'chunkhash' : 'hash',
   compiler_inline_manifest: false,
-  compiler_fail_on_warning: false,
+  compiler_fail_on_warning: __TEST__ || __PROD__,
   compiler_lint: argv.lint !== false,
   compiler_quiet: false,
   compiler_output_path: paths.base(config.dir_docs_dist),
-  compiler_public_path: __BASE__,
+  compiler_public_path: __BASE__ || '/',
```

Comment:
Removed the /config directory and 5 files of complexity in favor of 4 lines of conditionals in a single file. This caught duplicates and unused config as well.